lighty-modules/lighty-gnmi/lighty-gnmi-commons/src/main/java/io/lighty/modules/gnmi/commons/util/JsonUtils.java

```diff
@@ -28,10 +29,29 @@ private JsonUtils() {
     }
 
     public static String wrapJsonWithArray(final String jsonString, final String wrapper, final Gson gson,
-            final NodeIdentifierWithPredicates predicates) {
+            final NodeIdentifierWithPredicates predicates, final EffectiveModelContext context) {
         final JsonObject innerJson = JsonParser.parseString(jsonString).getAsJsonObject();
         for (final Entry<QName, Object> key : predicates.entrySet()) {
-            innerJson.add(key.getKey().getLocalName(), gson.toJsonTree(key.getValue()));
+            final var keyValue = key.getValue();
+            // InstanceIdentifier for identityref is stored as a QName, value should be in format MODULE:IDENTITY_NAME.
+            if (keyValue instanceof QName qnameValue) {
+                final var module = context.findModule(qnameValue.getModule()).orElseThrow();
+                final var value = String.format("%s:%s", module.getName(), qnameValue.getLocalName());
+                innerJson.add(key.getKey().getLocalName(), gson.toJsonTree(value));
+            // Custom ODL Number types are not correctly parsed by Gson.
+            } else if (keyValue instanceof Number numberValue) {
+                final var longValue = numberValue.longValue();
```

Comment:
Can't we use `instanceof` or something similar?

Reply:
IMO use `numberValue instanceof double` and `numberValue instanceof long`.

Reply:
KeyValue is set here:
https://github.com/PANTHEONtech/lighty/pull/1759/commits/ae0539e33d07358da6ca7818d31762ccff335d62#diff-779afd3e1579082e1dab107038965cfefcefab7eeddc4d03f440ee0dfc517d18R406-R418

So if keyValue is instanceof Number, it can only be one of the following: [DECIMAL64, Integer, Uint8, Uint16, Uint32 or Uint64].

I have used the method longValue() because it correctly displays every unit from the list [Integer, Uint8, Uint16, Uint32, Uint64], except for DECIMAL64. That's why "if (longValue == doubleValue) {" is included on line 46.

Reply:
Fixed with:
```
        if (numberValue instanceof Decimal64) {
            innerJson.add(key.getKey().getLocalName(), gson.toJsonTree(numberValue.doubleValue()));
        } else {
            innerJson.add(key.getKey().getLocalName(), gson.toJsonTree(numberValue.longValue()));
        }
```